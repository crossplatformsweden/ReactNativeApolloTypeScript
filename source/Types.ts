export enum NavigationConstants {
    /**
     * The login screen
     */
    LOGIN = 'LoginScreen',
    /**
     * The main screen
     */
    MAIN = 'MainScreen',
    /**
     *  The logged out screen
     */
    LOGOUT = 'LogoutScreen',
    /**
     * the map screen
     */
    MAP = 'MapScreen',
    /**
     * The profile screen
     */
    PROFILE = 'ProfileScreen',
}

export type RepoEdge = {
    edges: Array<RepoNode>;
};

export type Language = {
    name: string;
};

export type Owner = {
    login: string;
    url: string;
};

export type UserCount = {
    totalCount: number
};

export type RepoNodeItem = {
    id: string;
    name: string;
    url: string;
    descriptionHTML: string;
    primaryLanguage: Language;
    owner: Owner;
    stargazers: UserCount;
    viewerHasStarred: boolean;
    watchers: UserCount;
    viewerSubscription: string
};

export type RepoNode = {
    node: RepoNodeItem;
};

export type Starrable = {
    id: string;
    viewerHasStarred: boolean;
};

export type StarMutation = {
    starrable: Starrable;
};

export type AddStar = {
    addStar: StarMutation;
};

export type RemoveStar = {
    removeStar: StarMutation;
};

export type AddOrRemoeStar = AddStar | RemoveStar;

export type StarMutationData = {
    data: AddOrRemoeStar;
};

export type Viewer = {
    login: string;
    name: string;
    url: string;
    bio?: string;
    company?: any;
    repositories?: RepoEdge
};

export type ViewerResponse = {
    viewer?: Viewer;
};

export type Repository = {
    name: string,
    url: string
};

export type Organization = {
    name: string,
    url: string,
    repository?: Repository
};

export type OrganizationVariables = {
    organization: string, repository?: string
};

export type OrganizationResponse = {
    organization: Organization
};

export type SelectedRepositoryIdsResponse = {
    selectedRepositoryIds: Array<string>;
};

export type ToggleSelectedRepositoryVariables = {
    id: string;
    isSelected: boolean;
    __typename: string;
};

/**
 * Describes a generic GraphQL response where the data property is "any"
 */
export type GenericResponse = {
    loading: boolean;
    data?: any;
    error?: Error;
};

/**
 * Defines objects in application cache
 */
export interface IApplicationCache {
    selectedRepositoryIds: Array<string>;
}