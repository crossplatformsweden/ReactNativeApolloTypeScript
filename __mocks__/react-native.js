module.exports = {
  StyleSheet: {
    create: styles => styles,
  },
  Platform: {
    OS: 'ios',
  },
  Dimensions: {
    get: () => ({
      width: 1,
      height: 1,
    }),
  },
  Animated: {},
  AsyncStorage: {
    setItem: () => new Promise(resolve => resolve(), null),
    getItem: key => new Promise(resolve => resolve(`{"key": "${key}"}`), null),
    removeItem: (key, callback) => {
      if (callback) {
        callback();
      }
      return new Promise(resolve => resolve(), null);
    },
  },
};
