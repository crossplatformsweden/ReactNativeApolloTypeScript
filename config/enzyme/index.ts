import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Setup enzyme's react adapter
// @ts-ignore
Enzyme.configure({ adapter: new Adapter() });
