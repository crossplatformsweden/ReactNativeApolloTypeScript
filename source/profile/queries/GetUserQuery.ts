import RepositoryFragment from '../../repository/fragments/RepositoryFragment';
import gql from 'graphql-tag';

/**
 * Queries current user as "viewer" and repositories
 *
 * @see {RepositoryFragment} - referenced in this query
 */
const GetUserQuery = gql`
  {
    viewer {
    login,
    name,
    url,
    bio,
    company
    repositories(
        first: 5
        orderBy: { direction: DESC, field: STARGAZERS }
        ) {
            edges {
                node {
                    ...repository
                }
            }
        }
    }
  }

  ${RepositoryFragment}
`;

export default GetUserQuery;