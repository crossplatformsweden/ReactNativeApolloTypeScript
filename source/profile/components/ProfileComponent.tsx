import * as React from 'react';
import { View, Text } from 'react-native';
import Theme from '../../Theme';
import { Query } from 'react-apollo';
import { FormValidationMessage } from 'react-native-elements';
import BusyIndicator from '../../core/components/BusyIndicator';
import * as types from '../../Types';
import RepositoryList from '../../repository/components/RepositoryList';
import GetUserQuery from '../queries/GetUserQuery';

/**
 * Profile page pure React component
 * @typedef ProfilePure
 * @param data {@see ViewerResponse} resulting data from query
 */
export const ProfilePure = ({viewer}: types.ViewerResponse) =>
    (<View style={[Theme.container, Theme.paddingDefault]}>
        <Text style={Theme.title}>User</Text>
        <Text style={Theme.textBlock}>{viewer.name}</Text>
        <Text style={Theme.title}>Company</Text>
        <Text style={Theme.textBlock}>{viewer.company}</Text>
        <Text style={Theme.title}>Bio</Text>
        <Text style={Theme.textBlock}>{viewer.bio}</Text>
        <Text style={Theme.title}>Repositories</Text>
        <View style={[Theme.container, Theme.paddingDefault]}>
            <RepositoryList repositories={viewer.repositories} />
        </View>
    </View>);

/**
 * GraphQL Profile component querying user in GitHub API
 *
 * @typedef ProfileComponent
 * Wraps {@see ProfilePure}
 */
export const ProfileComponent = () => (
    <Query query={GetUserQuery}>
        {({ loading, data, error }: { loading: boolean, data: types.ViewerResponse, error?: Error }) => {
            if (error) {
                return <FormValidationMessage>{error.message}</FormValidationMessage>;
            }
            if (loading) {
                return <BusyIndicator isBusy message='Looking up user...' />;
            }

            if (!data || !data.viewer) {
                return null;
            }

            return (
                <ProfilePure {...data} />
            );
        }
        }
    </Query>
);

export default ProfileComponent;