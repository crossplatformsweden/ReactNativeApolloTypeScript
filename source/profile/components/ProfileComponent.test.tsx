import * as React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import { Query } from 'react-apollo';

import { ProfilePure } from './ProfileComponent';
import GetUserQuery from '../queries/GetUserQuery';
import { ViewerResponse } from '../../Types';
import { Text } from 'react-native';

    jest.unmock('react-native');

describe('ProfileComponent', () => {

    test('renders with data', () => {
        const data: ViewerResponse = {
            viewer: {
                name: 'nameMock',
                login: 'loginMock',
                bio: '',
                company: null,
                repositories: null,
                url: '',
            },
        };

        const wrapper = shallow(<ProfilePure {...data} />);

        expect(wrapper).toMatchSnapshot();

        const userNameElement = wrapper.find(Text).at(1);

        const nameText = userNameElement.render().text();
        expect(nameText).toEqual(data.viewer.name);

        const bioElement = wrapper.find(Text).at(5);
        const bioText = bioElement.render().text();
        console.log(bioText);
        expect(bioText).toEqual('');
    });
});