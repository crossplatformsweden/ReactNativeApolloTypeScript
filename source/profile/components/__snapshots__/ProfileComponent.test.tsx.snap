// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileComponent renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfilePure
    viewer={
      Object {
        "bio": "",
        "company": null,
        "login": "loginMock",
        "name": "nameMock",
        "repositories": null,
        "url": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 300,
            }
          }
        >
          User
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "width": 300,
            }
          }
        >
          nameMock
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 300,
            }
          }
        >
          Company
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "width": 300,
            }
          }
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 300,
            }
          }
        >
          Bio
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "width": 300,
            }
          }
        >
          
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 300,
            }
          }
        >
          Repositories
        </Text>,
        <Component
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "padding": 10,
              },
            ]
          }
        >
          <RepositoryList
            repositories={null}
          />
        </Component>,
      ],
      "style": Array [
        Object {
          "flex": 1,
        },
        Object {
          "padding": 10,
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "User",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "width": 300,
          },
        },
        "ref": null,
        "rendered": "User",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "nameMock",
          "ellipsizeMode": "tail",
          "style": Object {
            "width": 300,
          },
        },
        "ref": null,
        "rendered": "nameMock",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Company",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "width": 300,
          },
        },
        "ref": null,
        "rendered": "Company",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": null,
          "ellipsizeMode": "tail",
          "style": Object {
            "width": 300,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Bio",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "width": 300,
          },
        },
        "ref": null,
        "rendered": "Bio",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "",
          "ellipsizeMode": "tail",
          "style": Object {
            "width": 300,
          },
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Repositories",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "width": 300,
          },
        },
        "ref": null,
        "rendered": "Repositories",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <RepositoryList
            repositories={null}
          />,
          "style": Array [
            Object {
              "flex": 1,
            },
            Object {
              "padding": 10,
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "repositories": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "width": 300,
              }
            }
          >
            User
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "width": 300,
              }
            }
          >
            nameMock
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "width": 300,
              }
            }
          >
            Company
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "width": 300,
              }
            }
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "width": 300,
              }
            }
          >
            Bio
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "width": 300,
              }
            }
          >
            
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "width": 300,
              }
            }
          >
            Repositories
          </Text>,
          <Component
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "padding": 10,
                },
              ]
            }
          >
            <RepositoryList
              repositories={null}
            />
          </Component>,
        ],
        "style": Array [
          Object {
            "flex": 1,
          },
          Object {
            "padding": 10,
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "User",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 300,
            },
          },
          "ref": null,
          "rendered": "User",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "nameMock",
            "ellipsizeMode": "tail",
            "style": Object {
              "width": 300,
            },
          },
          "ref": null,
          "rendered": "nameMock",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Company",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 300,
            },
          },
          "ref": null,
          "rendered": "Company",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": null,
            "ellipsizeMode": "tail",
            "style": Object {
              "width": 300,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Bio",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 300,
            },
          },
          "ref": null,
          "rendered": "Bio",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "",
            "ellipsizeMode": "tail",
            "style": Object {
              "width": 300,
            },
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Repositories",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "width": 300,
            },
          },
          "ref": null,
          "rendered": "Repositories",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <RepositoryList
              repositories={null}
            />,
            "style": Array [
              Object {
                "flex": 1,
              },
              Object {
                "padding": 10,
              },
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "repositories": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
